File: AssayDashboard/Source/Hologic.AssayDashboard.AssayParser/Models/DataReduction.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Xml.Serialization;

namespace Hologic.AssayDashboard.AssayParser
{
    [XmlRoot("DataReductionMethod")]
    public class DataReduction
    {
        [XmlArray("ResultCategories")]
        public List<ResultCategory> resultCategories;

        [XmlElement("DataReductionAssembly")]
        public string reductionAssembly;

        [XmlArray("AssayFlags")]
        public List<AssayFlag> assayFlags;

        [XmlArray("AssaySettings")]
        public List<Setting> assaySettings;

        [XmlArray("AssayDilutions")]
        public List<AssayDilution> assayDilutions;
    }
}

File: AssayDashboard/Source/Hologic.AssayDashboard.Client/src/app/assayviewrepo/assayviewrepo-models/datareduction.ts
/*
// ----------------------------------------------------------------
// Copyright (c) 2017 Hologic, Inc.  All Rights Reserved.
// ----------------------------------------------------------------
*/

import { ResultCategory } from "app/assayviewrepo/assayviewrepo-models/resultcategory";
import { AssayDilution } from "app/assayviewrepo/assayviewrepo-models/assaydilution";
import { Setting } from "app/assayviewrepo/assayviewrepo-models/assaysetting";
import { AssayFlag } from "app/assayviewrepo/assayviewrepo-models/assayflag";

export class DataReduction {
    reductionAssembly: string;
    resultCategories: ResultCategory[];
    assayFlags: AssayFlag[];
    assaySettings: Setting[];
    assayDilutions: AssayDilution[];

    constructor() {
        this.resultCategories = undefined;
    }
}
